Fatras/include/ActsFatras/Kernel/InteractionList.hpp

```diff
@@ -0,0 +1,344 @@
+/// @note If an interaction destroys the incoming particle, the process
+///   simulation should indicate this via the break condition only and not
+///   by reducing the particle momentum to zero. The incoming particle should
+///   retain its initial kinematic state; the final kinematic state before
+///   destruction is typically of more interest to the user and this simplifies
+///   validation.
+///
+/// The physics processes are extendable by the user to accomodate their
+/// specific requirements. While the set of available physics processes must be
+/// configured at compile-time, within that set, processes can again be
+/// selectively disabled at run-time. By default all processes are applied.
+template <typename... processes_t>
+class InteractionList {
+  using Mask = std::bitset<sizeof...(processes_t)>;
+  using Processes = std::tuple<processes_t...>;
+  using ContinuousIndices = detail::ContinuousIndices<Processes>;
+  using PointLikeIndices = detail::PointLikeIndices<Processes>;
+
+ public:
+  /// Point-like interaction selection.
+  struct Selection {
+    Particle::Scalar x0Limit =
+        std::numeric_limits<Particle::Scalar>::infinity();
+    Particle::Scalar l0Limit =
+        std::numeric_limits<Particle::Scalar>::infinity();
+    size_t x0Process = SIZE_MAX;
+    size_t l0Process = SIZE_MAX;
+  };
+
+  /// Disable a specific process identified by index.
+  void disable(size_t process) { m_mask.set(process); }
+  /// Disable a specific process identified by type.
+  ///
+  /// @note Disables only the first element, if multiple elements of the same
+  ///   type exist.
+  template <typename process_t>
+  void disable() {
+    m_mask.set(detail::TupleIndexOf<process_t, Processes>::value);
+  }
+
+  /// Access a specific process identified by index.
+  template <size_t kProcess>
+  std::tuple_element_t<kProcess, Processes>& get() {
+    return std::get<kProcess>(m_processes);
+  }
+  /// Access a specific process identified by type.
+  ///
+  /// @warning This function only works if all configured processes have
+  ///   different types.
+  template <typename process_t>
+  process_t& get() {
+    return std::get<process_t>(m_processes);
+  }
+
+  /// Simulate the combined effects from all continuous interactions.
+  ///
+  /// @tparam generator_t must be a RandomNumberEngine
+  /// @param[in]     rng       is the random number generator
+  /// @param[in]     slab      is the passed material
+  /// @param[in,out] particle  is the particle being updated
+  /// @param[out]    generated is the container of generated particles
+  /// @return Break condition, i.e. whether a process stoped the propagation
+  template <typename generator_t>
+  bool runContinuous(generator_t& rng, const Acts::MaterialSlab& slab,
+                     Particle& particle,
+                     std::vector<Particle>& generated) const {
+    return runContinuousImpl(rng, slab, particle, generated,
+                             ContinuousIndices());
+  }
```

Comment:
Not sure I understand why these need separate `Impl` versions. Is it necessary for `ContinuousIndices` and `PointLikeIndices` to be an argument in the `Impl` methods.